File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Cpp/main.cpp
#include "./include/LenovoDesktopBuilder.hpp"
#include "./include/AsusDesktopBuilder.hpp"
#include "./include/DesktopDirector.hpp"

int main() {
    LenovoDesktopBuilder* lenovoDesktopBuilder = new LenovoDesktopBuilder();
    AsusDesktopBuilder* asusDesktopBuilder = new AsusDesktopBuilder();

    DesktopDirector* director1 = new DesktopDirector(lenovoDesktopBuilder);
    DesktopDirector* director2 = new DesktopDirector(asusDesktopBuilder);

    Desktop* desktop1 = director1->BuildDesktop();
    Desktop* desktop2 = director2->BuildDesktop();

    desktop1->showSpecifications();
    desktop2->showSpecifications();

    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Java/main.java
public class Main {
    public static void main(String[] args) {
        DesktopBuilder lenovoBuilder = new LenovoDesktopBuilder();
        DesktopBuilder asusBuilder = new AsusDesktopBuilder();

        DesktopDirector director1 = new DesktopDirector(lenovoBuilder);
        DesktopDirector director2 = new DesktopDirector(asusBuilder);

        Desktop lenovoDesktop  = director1.buildDesktop();
        Desktop asusDesktop  = director2.buildDesktop();

        lenovoDesktop .showSpecifications();
        asusDesktop .showSpecifications();
    }
}
